ggml-sycl.cpp

```diff
@@ -17419,6 +17414,7 @@ GGML_API GGML_CALL void ggml_backend_sycl_set_mul_device_mode() {
     g_sycl_gpu_mgr = new sycl_gpu_mgr();
     g_ggml_sycl_backend_gpu_mode = SYCL_MUL_GPU_MODE;
     ggml_init_by_gpus(g_sycl_gpu_mgr->get_gpu_count());
+    ggml_sycl_set_main_device(0);
```

Comment:
In mulitple mode, set main gpu is not needed. #0 gpu is always default main gpu.

Reply:
Unfortunately this isn't the case:

https://github.com/ggerganov/llama.cpp/blob/172c8256840ffd882ab9992ecedbb587d9b21f15/ggml-sycl.cpp#L3374

We could change the initial value of `g_main_device` from -1 to 0, but we'd probably also want to change some other initial values to stay in sync, say for `g_main_device_id`.  It seems safer to me to just call `ggml_sycl_set_main_device(0)` instead, but let me know what you prefer.